cmd/veneur-prometheus/main.go

```diff
@@ -39,12 +41,26 @@ func main() {
 		}).Fatal("failed to parse interval")
 	}
 
-	statsClient, _ := statsd.New(*statsHost, statsd.WithoutTelemetry())
+	var statsClient *statsd.Client
+
+	for {
```

Comment:
Would be good to at least have a comment here why we need this "endless" loop I think

Reply:
Also it could make sense to make this behaviour optional (i.e. have a separate flag for example).

Reply:
Makes sense. Will do.

Reply:
May also be worth considering adding some sort of timeout (30s)? Although if it makes it overcomplicated, can skip it.

General concern would be that the container will just seem to be hanging forever if it can't connect.

Reply:
I'm worried that if we have a timeout, kube might keep trying to restart the container, and helm might recognise the deployment as bas. I'm thinking since it's not a compulsory container, it might be okay if it keeps hanging? We can set an alarm if we lose metrics for an extended period of time.